problem-gson/src/main/java/org/zalando/problem/DefaultProblemAdapter.java

```diff
@@ -0,0 +1,150 @@
+package org.zalando.problem;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.internal.bind.TypeAdapters;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonToken;
+import com.google.gson.stream.JsonWriter;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+final class DefaultProblemAdapter extends TypeAdapter<Problem> {
+
+    private final Gson gson;
+    private final boolean stacktrace;
+    private TypeAdapter<URI> uriAdapter;
+    private TypeAdapter<StatusType> statusAdapter;
+    private TypeAdapter<ThrowableProblem> throwableProblem;
+
+    DefaultProblemAdapter(Gson gson, boolean stacktrace) {
+        uriAdapter = gson.getAdapter(URI.class);
+        statusAdapter = gson.getAdapter(StatusType.class);
+        throwableProblem = gson.getAdapter(ThrowableProblem.class);
+        this.gson = gson;
+        this.stacktrace = stacktrace;
+    }
+
+    @Override
+    public void write(JsonWriter out, @Nullable Problem problem) throws IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'write(JsonWriter, Problem)' has an NPath complexity of 1728](https://app.codacy.com/app/whiskeysierra/problem/pullRequest?prid=4476243)